### PR TITLE
API Server - rebased on 1.6 vendor changes, and compiling main.go

### PR DESCRIPTION
**Pull request by kris-nova**
I think this is just 2 commits ahead of https://github.com/kubernetes/kops/pull/2184, but is also rebased on to kops `head`

This PR offers the working dependency changes we need, and gets our work back up to speed to where it was before the huge vendor changes of 1.6

Offering this PR as an **alternative** to https://github.com/kubernetes/kops/pull/2184 so spare the nasty double merge conflict resolution that would need to happen in @justinsb's branch to accept mine.

I also opened up https://github.com/justinsb/kops/pull/5 that does just that if that is the route we want to take!

I would *love* to get this merged in ASAP as this has 0 impact on `kops` as it stands today.. but having this in master gives @justinsb and myself a clean starting point for iterating on the smaller features. 

Example:  I still need to fix the etcd annotations in the helm chart.

TLDR;

This is exactly the same as https://github.com/kubernetes/kops/pull/2184 but fixes the dependency issues and gets `main.go` compiling again for the server. I think we need to merge this soon (@justinsb would totes agree). This doesn't touch existing `kops` code at all. Please merge.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/2204)

### CODE REVIEW

File: docs/api-server/README.md [hunk 1]
```diff
@@ -1,6 +1,19 @@
 # Kops HTTP API Server
 
-### Notes:
+### Building
 
+To build the API container run the following
 
- - https://github.com/kubernetes/kubernetes/pull/40803
\ No newline at end of file
+Note: This is a MAJOR clockwork, and a lot of assumptions and hardcoding exist here for now.
```
Comment (justinsb) [inline docs/api-server/README.md]: BTW I am not familiar with the term "clockwork"

**Comment by justinsb** [comment]
I think this LGTM.  Some of the commits are mine obviously, but if you also think this LGTM we meet the "no self approval" rule between us... feel free to merge at will.  I think you do need to run `make copydeps` though, to add the new dependencies to `vendor`

**Comment by kris-nova** [comment]
Your commits are still in the logs here, let me run make copydeps, update the docs and make sure I can get this running with helm

But thanks for the LGTM

File: docs/api-server/README.md (hunk 1 above)
Comment (kris-nova) [inline docs/api-server/README.md]: As in "clockwork orange" 

"This PR is a clockwork" ~> "This PR is a work in progress"
